app/src/main/java/ch/sdp/vibester/activity/BuzzerSetupActivity.kt

```diff
@@ -0,0 +1,96 @@
+package ch.sdp.vibester.activity
+
+import android.content.Context
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.view.View
+import android.widget.*
+import androidx.core.view.children
+import ch.sdp.vibester.R
+
+/**
+ * Class to set up buzzer game (number of players)
+ */
+class BuzzerSetupActivity : AppCompatActivity(), AdapterView.OnItemSelectedListener {
+    var text = "One"
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_buzzer_setup)
+        val ctx: Context = this
+
+        val spinner: Spinner = findViewById(R.id.nb_player_spinner)
+        val adapter = ArrayAdapter.createFromResource(
+            ctx,
+            R.array.nb_players,
+            android.R.layout.simple_spinner_item
+        )
+        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
+        spinner.onItemSelectedListener = this
+    }
+
+    override fun onItemSelected(parent: AdapterView<*>, view: View?, position: Int, id: Long) {
+        text = parent.getItemAtPosition(position).toString()
+        updatePlayerNameVisibility(textToNumber(text), R.id.namePlayer2)
+        updatePlayerNameVisibility(textToNumber(text), R.id.namePlayer3)
+        updatePlayerNameVisibility(textToNumber(text), R.id.namePlayer4)
+    }
+
+    /**
+     * Converts the spinner text for the number of players into an Int
+     * @param
+     * text: the string to be converted
+     */
+    fun textToNumber(text: String): Int {
+        when (text) {
+            "One" -> return 1
+            "Two" -> return 2
+            "Three" -> return 3
+            "Four" -> return 4
+        }
+        return 1
```

Comment:
Is it possible to test the return value?

Reply:
Done!